File: templategenerator.systemtest/src/test/java/com/optivem/atddaccelerator/templategenerator/systemtest/dsl/GeneratorDsl.java
package com.optivem.atddaccelerator.templategenerator.systemtest.dsl;

import com.optivem.atddaccelerator.templategenerator.systemtest.clients.GeneratorClient;

import java.util.HashSet;
import java.util.Set;

import static com.optivem.atddaccelerator.templategenerator.systemtest.util.ProcessExecutor.waitTime;
import static com.optivem.atddaccelerator.templategenerator.systemtest.util.ProcessResultAssertions.assertFailure;
import static com.optivem.atddaccelerator.templategenerator.systemtest.util.ProcessResultAssertions.assertSuccess;

public class GeneratorDsl {
    private static final int MILLIS = 3000;

    private Set<String> created;

    private final GeneratorClient client;

    public GeneratorDsl(GeneratorClient client) {
        this.client = client;
        this.created = new HashSet<>();
    }

    public boolean isCreated(String repoName) {
        return created.contains(repoName);
    }

    public void generateNewRepository(String repoName, String systemLanguage, String systemTestLanguage) {
        var result = client.generateRepository(repoName, systemLanguage, systemTestLanguage);
        assertSuccess(result);

        waitTime(MILLIS);
        created.add(repoName);
    }

    public void generateNewRepositoryExpectError(String repoName, String systemLanguage, String systemTestLanguage) {
        var result = client.generateRepository(repoName, systemLanguage, systemTestLanguage);
        assertFailure(result);
    }
}

File: console/tests/Dsl/GeneratorDsl.cs
using Optivem.AtddAccelerator.TemplateGenerator.SystemTests.Clients;
using Optivem.AtddAccelerator.TemplateGenerator.SystemTests.Util;
using static Optivem.AtddAccelerator.TemplateGenerator.SystemTests.Util.Process.ProcessExecutor;
using static Optivem.AtddAccelerator.TemplateGenerator.SystemTests.Util.Process.ProcessResultAssertions;

namespace Optivem.AtddAccelerator.TemplateGenerator.SystemTests.Dsl
{
    public class GeneratorDsl
    {
        private const int Millis = 3000;

        private readonly HashSet<string> _created;
        private readonly GeneratorClient _client;

        public GeneratorDsl(GeneratorClient client)
        {
            _client = client;
            _created = new HashSet<string>();
        }

        public bool IsCreated(string repoName)
        {
            return _created.Contains(repoName);
        }

        public void GenerateNewRepository(string repoName, Language systemLanguage, Language systemTestLanguage)
        {
            var result = _client.GenerateRepository(repoName, systemLanguage, systemTestLanguage);
            AssertSuccess(result, "Repository generation should succeed.");

            WaitTime(Millis);
            _created.Add(repoName);
        }

        public void GenerateNewRepositoryExpectError(string repoName, Language systemLanguage, Language systemTestLanguage)
        {
            var result = _client.GenerateRepository(repoName, systemLanguage, systemTestLanguage);
            AssertFailure(result, "Repository generation should fail.");
        }
    }
}